pkg/workloadapi/workload.proto

```diff
@@ -154,11 +154,12 @@ message Workload {
   // WorkloadName represents the name for the workload (of type WorkloadType). Used for telemetry.
   string workload_name = 13;
 
+  // Deprecated: superseded by native_tunnel.
+  // If set, it's equivilant to using native_tunnel with HBONE protocol.
+  bool is_native_tunnel = 14;
+
   // If set, this indicates a workload expects to directly receive tunnel traffic.
-  // In ztunnel, this means:
-  // * Requests *from* this workload do not need to be tunneled if they already are tunneled by the tunnel_protocol.
-  // * Requests *to* this workload, via the tunnel_protocol, do not need to be de-tunneled.
-  bool native_tunnel = 14;
+  NativeTunnel native_tunnel = 23;
```

Comment:
Naming bikeshed: something like `application_protocol` seems to fit better? Or `application_tunnel_protocol`

Reply:
Then below the enum is ApplicationTunnelProtocol

Reply:
+1 - native doesn't mean much, and many things are a tunnel

tunnel_protocol or application_tunnel_protocol are a bit more clear.

Reply:
+1 native is confusing